pkg/internal/infrastructure/terraform.go

```diff
@@ -171,6 +176,38 @@ func ComputeTerraformerChartValues(infra *extensionsv1alpha1.Infrastructure, cli
 	}, nil
 }
 
+func getNatGatewayValues(natConfig *api.NatGatewayConfig) *map[string]interface{} {
+	if natConfig == nil || !natConfig.Enabled {
+		return nil
+	}
+	if len(natConfig.IPAddresses) == 0 && len(natConfig.IPAddressRanges) == 0 {
+		return nil
+	}
+	var values = map[string]interface{}{}
+	if ipAddresses := resourceReferencesToValue(&natConfig.IPAddresses); ipAddresses != nil {
+		values["ipAddresses"] = *ipAddresses
+	}
+	if ipAddressRanges := resourceReferencesToValue(&natConfig.IPAddressRanges); ipAddressRanges != nil {
+		values["ipAddressRanges"] = *ipAddressRanges
+	}
+	return &values
+}
+
+func resourceReferencesToValue(references *[]api.AzResourceReference) *[]map[string]interface{} {
+	var count = len(*references)
+	if count == 0 {
+		return nil
+	}
+	var result = make([]map[string]interface{}, count)
+	for i, ref := range *references {
+		result[i] = map[string]interface{}{
```

Comment:
```suggestion
		result = append(result, map[string]interface{}{
```

Reply:
If we do that we would need to init the slice with length 0 as append expand the slice. In combination with `make([]map[string]interface{}, count)` we would get a length of `2x count` and a slice with empty values.

So with append we can use `var result = []map[string]interface{}{}` to init the slice.

But I prefer if the size is already known to init the slice with required size and require the amount of memory directly. Then no copying during appends would be required.